From b11205d1c0438eee653d7a327a343e5deeaf6144 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Wed, 25 May 2022 20:26:34 +0200
Subject: [PATCH 333/417] media: platform: rkisp1: Fix detection of sensor
 subdevices

Detect by inclusion of the sensor subdev, not by exclusion of all
subdevices known to rkisp1.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/media/platform/rockchip/rkisp1/rkisp1-dev.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/media/platform/rockchip/rkisp1/rkisp1-dev.c b/drivers/media/platform/rockchip/rkisp1/rkisp1-dev.c
index 3f5cfa7eb..ee76808ed 100644
--- a/drivers/media/platform/rockchip/rkisp1/rkisp1-dev.c
+++ b/drivers/media/platform/rockchip/rkisp1/rkisp1-dev.c
@@ -129,9 +129,7 @@ static int rkisp1_create_links(struct rkisp1_device *rkisp1)
 	/* sensor links */
 	flags = MEDIA_LNK_FL_ENABLED;
 	list_for_each_entry(sd, &rkisp1->v4l2_dev.subdevs, list) {
-		if (sd == &rkisp1->isp.sd ||
-		    sd == &rkisp1->resizer_devs[RKISP1_MAINPATH].sd ||
-		    sd == &rkisp1->resizer_devs[RKISP1_SELFPATH].sd)
+		if (sd->entity.obj_type != MEDIA_ENTITY_TYPE_V4L2_SUBDEV || sd->entity.function != MEDIA_ENT_F_CAM_SENSOR)
 			continue;
 
 		ret = media_entity_get_fwnode_pad(&sd->entity, sd->fwnode,
-- 
2.35.3

